services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_insights/history.tsx

```diff
@@ -0,0 +1,201 @@
+  const [loading, setLoading] = React.useState<boolean>(true);
+  const [studentHistories, setStudentHistories] = React.useState<StudentHistoryInterface[]>(null)
+
+  React.useEffect(() => {
+    setLoading(true)
+    const queryHash = { timeframe }
+    if (useELLDiagnostics) {
+      queryHash['ell'] = true
+    }
+    const url = queryString.stringifyUrl({ url: `/diagnostic_insights/history_by_classroom/${classroom.id}`, query: queryHash }, { arrayFormat: 'bracket' })
+
+    requestGet(url, (body) => {
+      setStudentHistories(body)
+      setLoading(false)
+    })
+  }, [classroom, timeframe, useELLDiagnostics])
+
+  if (loading) {
+    return (
+      <div className="diagnostic-insights-content history by-class">
+        <Spinner />
+      </div>
+    )
+  }
+
+
+  const renderDiagnosticSquares = (studentLearningSequences: StudentLearningSequenceInterface[]) => {
+    if (!studentLearningSequences) { return null }
+
+    return studentLearningSequences.map(studentLearningSequence => {
+      const { student_learning_sequence_activities, } = studentLearningSequence
+      const { pre_diagnostic, post_diagnostic, pre_recommendation, post_recommendation, } = student_learning_sequence_activities
+
+      // sometimes there is a studentLearningSequence when a student has completed an activity that belongs to the recommendation sequence,
+      // even if they didn't complete either of the diagnostics, in which case we don't want to render anything
+      if (!pre_diagnostic) { return null }
+
+      let icon: JSX.Element
+
+      if (pre_diagnostic && post_diagnostic) {
+        icon = pre_diagnostic[0].proficient || post_diagnostic[0].proficient ? filledBlueDiamondWithCheckMarkImg : filledBlueDiamondImg
+      } else {
+        icon = pre_diagnostic[0].proficient ? outlinedBlueDiamondWithCheckMarkImg : outlinedBlueDiamondImg
+      }
+
+      const tooltipText = []
+
+      // Baseline Pre Completed
+      if (pre_diagnostic?.length) {
+        const preDiagnosticCompletionDate = moment(pre_diagnostic[pre_diagnostic.length - 1].completed_at).format("MM/DD/YY");
+        tooltipText.push(
+          <p>
+            <strong>Baseline Pre Completed:</strong> {preDiagnosticCompletionDate}
+          </p>
+        );
+      }
+
+      // Baseline Pre Practice
+      if (pre_recommendation?.length) {
+        const firstPracticeDate = moment(pre_recommendation[0].completed_at).format("MM/DD/YY")
+        const lastPracticeDate = moment(pre_recommendation[pre_recommendation.length - 1].completed_at).format("MM/DD/YY")
+        tooltipText.push(
+          <p>
+            <strong>Baseline Pre Practice:</strong> {pre_recommendation.length} activities (
+            {firstPracticeDate} - {lastPracticeDate})
+          </p>
+        );
+      }
+
+      // Growth Post Completed
+      if (post_diagnostic?.length) {
+        const postDiagnosticCompletionDate = moment(post_diagnostic[post_diagnostic.length - 1].completed_at).format("MM/DD/YY")
```

Comment:
`moment` is in [maintenance mode til EOL](https://momentjs.com/docs/#/-project-status/). 

I think we should not add momentjs code to any codebases, and instead use the builtins or leaner, maintained projects like luxon. Happy to debate this though.

Reply:
Good call-out. I'm replacing it with a built-in and standardizing it as a function since it's less succinct than the moment formatting.